extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/group/UserGroupService.java

```diff
@@ -165,14 +165,44 @@ private String getGroupSearchFilter() throws GuacamoleException {
         if (groupBaseDN == null)
             return Collections.emptyList();
 
+        // memberAttribute specified in properties could contain DN or username 
+        String memberAttributeType = confService.getMemberAttributeType();
+        String userID = userDN;
+        switch (memberAttributeType) {
```

Comment:
Might be worth doing a new `enum` type for this and a `GuacamoleProperty` that validates it.

Reply:
Looks much better with the `enum`, here, but, now I think we can probably get rid of the `switch()` statement, no?  Based on the way the logic is constructed, here, I think you can just do the `String userID = userDN;` assignment, and then do a `if (memberAttributeType == MemberAttributeType.UID)` with and re-assign as necessary?